charm/ksql/lib/charms/layer/ksql.py

```diff
@@ -0,0 +1,177 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ipaddress
+import os
+import netifaces
+import re
+import shutil
+import socket
+import yaml
+from pathlib import Path
+
+from charmhelpers.core import hookenv, host
+from charmhelpers.core.templating import render
+
+KSQL_PORT=8088
+KSQL_SNAP='ksql-server'
+KSQL_SERVICE='snap.{}.ksql-server.service'.format(KSQL_SNAP)
+# TODO: Change this to common, once the new ksql server snap is published
+KSQL_SNAP_COMMON='/var/snap/{}/current'.format(KSQL_SNAP)
+KSQL_CLUSTER_ID='production'
+
+class Ksql(object):
+    def open_ports(self):
+        hookenv.open_port(KSQL_PORT)
+    
+    def close_ports(self):
+        hookenv.close_port(KSQL_PORT)
+
+    def configure(self, kafka_units, network_interface=None):
+        kafka = []
+        for unit in kafka_units:
+            ip = resolve_private_address(unit['host'])
+            kafka.append('{}:{}'.format(ip, unit['port']))
+        kafka.sort()
+        kafka_connect = ','.join(kafka)
+
+        ip = get_ip_for_interface(network_interface) if network_interface else hookenv.unit_private_ip()
```

Comment:
Let's simplify this. Remove the network_interface argument, as it's not used by the reactive layer (no config options). That makes `ip = hookenv.unit_private_ip()` and we can then also remove get_ip_for_interface, the netifaces import, and the python-netifaces package dependency up above in layer.yaml.

Less code means less to support and reason about.